framework/src/org/checkerframework/framework/util/AtmLubVisitor.java

```diff
@@ -115,14 +112,8 @@ private AnnotatedTypeMirror lubWithNull(
     private void lubPrimaryAnnotations(
             AnnotatedTypeMirror type1, AnnotatedTypeMirror type2, AnnotatedTypeMirror lub) {
         Set<? extends AnnotationMirror> lubSet;
-        if (visitingUninferedWildcard
-                        && type1.getAnnotations().size() != type1.getAnnotations().size()
```

Comment:
Should it have been "type*1*... != type*2*"?  Did you try that change?  From the surrounding code seems like the most likely typo, rather than the typo being that the condition shouldn't have been there at all.

@kelloggm You might want to chime in, and to create a test that relies on this behavior.